internal/cli/hclgen/hclgen.go

```diff
@@ -0,0 +1,659 @@
+						cont, err, close := populateField(sfield.Field, sfield.Type, ui)
+						if err != nil {
+							return m, err, false
+						} else if close {
+							return m, nil, true
+						}
+						m[field.Field].children[sfield.Field] = cont
+					}
+				}
+			} else {
+				cont, err, close := populateField(field.Field, field.Type, ui)
+				if err != nil {
+					return m, err, false
+				} else if close {
+					return m, nil, true
+				}
+				m[field.Field] = fieldInfo{contents: cont, isParent: false}
+			}
+		}
+	}
+	return m, nil, false
+}
+
+func populateField(name string, fType string, ui terminal.UI) (string, error, bool) {
+	getField := true
+	typeString := fType
+	if typeString == "" {
+		typeString = "No_Type_Specified"
+	}
+	for getField {
+		fieldVal, err := ui.Input(&terminal.Input{
+			Prompt: fmt.Sprintf("%s <%s>: ", strings.Title(name), typeString),
+			Style:  "",
+			Secret: false,
+		})
+		if err != nil {
+			ui.Output(
+				"Error getting input: %s",
+				clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+			return "", err, false
+		} else if strings.ToLower(fieldVal) == "exit" {
+			return "", nil, true
+		} else if strings.ToLower(fieldVal) == "" {
+			ui.Output(fmt.Sprintf("You have selected to skip the %s field.", name))
+			pNameConfirm, err := ui.Input(&terminal.Input{
+				Prompt: fmt.Sprintf("Do you really want to skip the %s field? (y/N): ", name),
+				Style:  "",
+				Secret: false,
+			})
+			if err != nil {
+				ui.Output(
+					"Error getting input: %s",
+					clierrors.Humanize(err),
+					terminal.WithErrorStyle(),
+				)
+				return "", err, false
+			} else if strings.ToLower(pNameConfirm) == "exit" {
+				return "", nil, true
+			} else if strings.ToLower(pNameConfirm) == "yes" || strings.ToLower(pNameConfirm) == "y" {
+				ui.Output("%s skipped\n", strings.Title(name), terminal.WithWarningStyle())
+				return "", nil, false
+			} else {
+				ui.Output("Skip cancelled\n")
+			}
+		} else {
+			// TODO: field input type checking
+			ui.Output("You inputted \"%s\"\n", fieldVal)
+			fieldConfirm, err := ui.Input(&terminal.Input{
+				Prompt: "Is this correct? (y/N): ",
+				Style:  "",
+				Secret: false,
+			})
+			if err != nil {
+				ui.Output(
+					"Error getting input: %s",
+					clierrors.Humanize(err),
+					terminal.WithErrorStyle(),
+				)
+				return "", err, false
+			} else if strings.ToLower(fieldConfirm) == "exit" {
+				return "", nil, true
+			} else if strings.ToLower(fieldConfirm) == "yes" || strings.ToLower(fieldConfirm) == "y" {
+				ui.Output("%s confirmed\n", strings.Title(name), terminal.WithSuccessStyle())
+				return fieldVal, nil, false
+			} else {
+				ui.Output("%s rejected\n", strings.Title(name))
+			}
+		}
+	}
+	return "", nil, true
+}
+
+// <plug> indicates the plugin that the user needs to select. 1: Builder, 2: Deployer/Platform, 3: Releaser, 4: Registry
+func selectPlugin(plug int, fList []string, fSystem embed.FS, ui terminal.UI) (PlugDocs, error, bool) {
```

Comment:
`plug int` could be simplified to be `plug string` and then the case statements would read

`case "builder":` ... `case "releaser":`